xrpl/token.js

```diff
@@ -0,0 +1,101 @@
+const { Command } = require('commander');
+const { hex } = require('./utils');
+const { printInfo, printError } = require('../common');
+
+function tokenSymbolToCurrencyCode(tokenSymbol) {
+    if (tokenSymbol.length <= 3) {
+        return tokenSymbol;
+    }
+
+    const hexString = hex(tokenSymbol).toUpperCase();
+
+    if (hexString.length > 40) {
+        printError(
+            `Token symbol "${tokenSymbol}" too long: hex representation (${hexString.length} chars) exceeds xrpl 40-character limit`,
+        );
+        return;
```

Comment:
throw an exception instead. printing an error but returning nothing in a function that has a return argument is bad practice. Also, let's use Typescript going forward for any new scripts.